### PR TITLE
maptrim app conversion + gtests

### PR DESCRIPTION
**Pull request by AustinSanders**
Converts maptrim from app to function and adds 5 gtests. Also includes fixes to fixtures + projected cube pvl.

## Description
App conversion + gtests.  Reopens projTestCube in defaultCube setup in order to save DNs to the cube.  Renames MissonPhaseName in projected cube pvl to MissionPhaseName.

## Related Issue
Testing

## Motivation and Context
273Mb

## How Has This Been Tested?
5 Gtests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by AustinSanders** [comment]
I modified projtestcube.pvl to include MissionPhaseName rather than MissonPhaseName + I added a reopen in the projTestCube setup in order to save the DNs.  I'm hoping this doesn't cause failures, but if we start seeing a ton of errors pop up, then that's probably why.

**Comment by AustinSanders** [comment]
It looks like some of the automos tests are now failing on the histogram tests due to the presence of DNs.  They used to be all NULLs, now they have values.  Should I edit those tests as part of this PR?

File: isis/src/base/apps/maptrim/maptrim.cpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+using namespace std;
+namespace Isis{
+
+  void trim(Buffer &in, Buffer &out);
+  void getSize(Buffer &in);
+
+  double slat, elat, slon, elon;
+  int smallestLine, biggestLine, smallestSample, biggestSample;
+  TProjection *proj;
```
Comment (Kelvinrr) [inline isis/src/base/apps/maptrim/maptrim.cpp]: I would make these static

File: isis/tests/FunctionalTestsAutomos.cpp [hunk 2]
```diff
@@ -49,10 +49,10 @@ TEST_F(DefaultCube, FunctionalTestAutomosDefault) {
...
   std::unique_ptr<Histogram> oCubeStats(mos.histogram());
 
-  EXPECT_NEAR(oCubeStats->Average(),           -1.7976931348623149e+308,    0.001);
-  EXPECT_NEAR(oCubeStats->Sum(),               0,  0.000001);
-  EXPECT_NEAR(oCubeStats->ValidPixels(),       0,     0.000001);
-  EXPECT_NEAR(oCubeStats->StandardDeviation(), -1.7976931348623149e+308, 0.001);
+  EXPECT_DOUBLE_EQ(oCubeStats->Average(), 123.5);
+  EXPECT_EQ(oCubeStats->Sum(), 4446);
+  EXPECT_EQ(oCubeStats->ValidPixels(), 36);
+  EXPECT_DOUBLE_EQ(oCubeStats->StandardDeviation(), 79.757668686375951);
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsAutomos.cpp:55]: Im surprised no one caught that the original tests and nulls and zeros

**Review by Kelvinrr — COMMENTED** [review]
were the original tests removed?

**Review by acpaquette — COMMENTED** [review]
Overall looks good! The only thing I think that could be added is a test that the applog has a pvlgroup added to it after a successful run.

**Review by acpaquette — APPROVED** [review]

**Pull request merged**